### PR TITLE
N°3300 - Add creation and modification of 1-n objects in object details

### PR DESCRIPTION
**Pull request by steffunky**
Allows for the creation and modification of **direct** links in popups when accessing an object in visualization.
#378 implementation for 1-n

### CODE REVIEW

**Comment by Molkobain** [comment]
Reviewed this morning with Stephen. Some stuff to rework but will be done on #378 as it contains most of the change. Once #378 is merged, we will rework history of this one and merge it as well.

**Comment by Hipska** [comment]
Sounds great, do you have a before and after screenshot please?

File: sources/Application/UI/Links/Direct/BlockDirectLinksViewTable.php [hunk 1]
```diff
@@ -30,14 +30,24 @@ public function GetTargetClass(): string
...
 			'object_id'   => $this->oDbObject->GetKey(),
 			'menu'        => MetaModel::GetConfig()->Get('allow_menu_on_linkset'),
 			'default'     => $this->GetDefault(),
 			'table_id'    => $this->GetTableId(),
 			'row_actions' => $this->GetRowActions(),
+			'currentId' => $this->GetTableId(),
 		);
+
+		// - Add creation in modal if the linkset is not readonly
+
```
Comment (Molkobain) [inline sources/Application/UI/Links/Direct/BlockDirectLinksViewTable.php:44]:
```suggestion
		// - Add creation in modal if the linkset is not readonly
```

File: sources/Application/UI/Links/Direct/BlockDirectLinksViewTable.php [hunk 2]
```diff
@@ -30,14 +30,24 @@ public function GetTargetClass(): string
...
+		if (!$this->oAttDef->GetReadOnly()) {
+			$aExtraParams['creation_in_modal_is_allowed'] = true;
+			$aExtraParams['creation_in_modal_js_handler'] = 'LinkSetWorker.CreateLinkedObject("'.$this->GetTableId().'");';
+		}
```
Comment (Molkobain) [inline sources/Application/UI/Links/Direct/BlockDirectLinksViewTable.php]:
```suggestion

		// - Add creation in modal if the linkset is not readonly
		if (!$this->oAttDef->GetReadOnly()) {
			$aExtraParams['creation_in_modal_is_allowed'] = true;
			$aExtraParams['creation_in_modal_js_handler'] = 'LinkSetWorker.CreateLinkedObject("'.$this->GetTableId().'");';
		}
```

**Review by Molkobain — APPROVED** [review]

**Pull request merged**